lib/ask/runtime/flow.ex

```diff
@@ -40,10 +40,20 @@ defmodule Ask.Runtime.Flow do
 
   defp next_step_by_skip_logic(flow, step, reply_value) do
     skip_logic =
-      step
-      |> Map.get("choices")
-      |> Enum.find(fn choice -> choice["value"] == reply_value end)
-      |> Map.get("skip_logic")
+      case step["type"] do
+        "numeric" ->
+          value = String.to_integer(reply_value)
+          step["ranges"]
+          |> Enum.find_value(nil, fn (range) ->
+            if (range["from"] == nil || range["from"] <= value) && (range["to"] >=
+              value || range["to"] == nil), do: range["skip_logic"], else: false
```

Comment:
There is a [test](https://github.com/instedd/ask/pull/363/files#diff-d5b2e2f1b6436a3d4a30d58988617b2bR126) for it. The problem is that comparing `nil` with an integer doesn't raise an exception, it compares bigger to all of them. I'll re order the formula clauses just for clarity since the check for `nil` on the `range["to"]` could be deleted all together due to this strange behavior.